shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
+                msg = "'additional_server_presets' parameter must be a list or None"
+                raise ValueError(msg)
+
+            preset_list = []
+            for preset in additional_server_presets:
+                if not isinstance(preset, list):
```

Comment:
For future code readers, adding a comment with an example of what the code is trying to parse would be helpful I think! E.g something along the lines of 
```
# now generate crud-ready filter preset
# ['name_of_preset', {'param1': 'value1', 'param2': 'value2'} ]
# ->
# {'name': 'name_of_preset', 'param1': 'value1', 'param2': 'value2'}
```